### ISSUE TITLE
Support for å lese flere tekster fra resource-fil inn i samme felt i appen

### ISSUE THREAD

**Issue by violaguttorm** [Altinn/app-frontend-react#1695]
I rrh-innrapportering har vi en tableHeader som vi definerer ut i fra verdien til flere forskjellige felter i datamodellen. Her er vi per nå nødt til å hardkode inn verdi at runtime i vår DataProcessingHandler når hvilket som helst av underfeltene oppdateres, og vi er også nødt til å hardkode inn selve teksten da jeg ikke har funnet noen måte å definere dette som en enkelt variabel i resource.nb.json fila. 

Ideelt sett skulle jeg sett en metode hvor vi f. eks kunne brukt dynamiske uttrykk til å flette sammen flere verdier fra resource.nb.json fil til et enkeltfelt, men en løsning hvor vi kan gjøre det samme i C# fungerer også. 

Legger ved et bilde av tableHeader, nåværende brukt for å generere tableHeader, og hva brukeren ville fylt ut for å generere akkurat denne tableHeaderen, håper det hjelper dersom beskrivelsen er dårlig forklart. 

![Image](https://github.com/Altinn/app-frontend-react/assets/57175682/35e7eda0-427f-4a46-b25a-b87dac9aa772)

**Comment by olemartinorg** [comment]
@violaguttorm Beklager sent svar her.

> Ideelt sett skulle jeg sett en metode hvor vi f. eks kunne brukt dynamiske uttrykk til å flette sammen flere verdier fra resource.nb.json fil til et enkeltfelt

Jeg mener [`concat`-funksjonen](https://docs.altinn.studio/nb/altinn-studio/reference/logic/expressions/#func-concat) gjør akkurat dette. Har du prøvd den ut og sett om den kan fungere?

Det blir et svært uttrykk til slutt, men jeg mener du skal kunne gjøre noe slikt for å gjenskape kodesnutten fra C# med et uttrykk:

```json
{
  "id": "something",
  "type": "SomeComponentType",
  "textResourceBindings": {
    "title": ["concat",
      ["if", ["dataModel", "harPosisjonEierskap.Value"], ["concat", "Eierskap ", ["dataModel", "posisjonEierskap.storrelsesintervall"]]],
      ["if", ...]
    ]
  }
}
```

**Comment by violaguttorm** [comment]
@olemartinorg Ja, concat gjør det jeg vil, men jeg trenger denne _ikke_ som en tekst, men som en verdi (under en tableHeader). Noe sånt som dette: 

```
{
	"id": "reelle-group",
	"type": "RepeatingGroup",
	"children": [
	  ... alle barna her
	],
	"tableHeaders": [
	  "rettighetshaver-fulltNavn-tableHeader",
	  [ "concat", "verdier fra både resource", "og datamodell" ]
	],
	... mer config
},
```

**Comment by violaguttorm** [comment]
... jeg ble forvirret av GUIet og skulle velge vekk "Close with comment" da jeg trodde det var en nedtrekksliste, ikke en knapp, og nå får jeg ikke åpnet den igjen. Kan du gjøre det? 😅

**Comment by olemartinorg** [comment]
Null problem - men jeg tror det kan være riktig å lukke den om behovene allerede er dekket. 😁

For det jeg tror mangler i puslespillet er at du prøver det ut med `Text`-komponenten: https://docs.altinn.studio/altinn-studio/reference/ux/components/text/

Der kan du bygge opp uttrykket i `value`-egenskapen, og putte hele komponenten i `RepeatingGroup`. Så selve uttrykket som produserer teksten passer altså ikke inn i `tableHeaders`, men i komponenten du har i den cella.

**Comment by violaguttorm** [comment]
Ja, Text gjør egentlig akkurat hva jeg er ute etter, og bekreftet at det funker! 🤩 Da kan saken lukkes (igjen). 🙂
